### PR TITLE
Cleanup in preparation for XEN_DOMCTL_createdomain changes upstream

### PR DESCRIPTION
**Pull request by andyhhp**

### CODE REVIEW

**Comment by andyhhp** [comment]
Actually in some cases, create_exn appears to not have build_info at the point it is wanted.  Perhaps because this is the first boot of the VM?

File: xc/xenops_server_xen.ml [hunk 1]
```diff
@@ -1095,7 +1095,12 @@ module VM = struct
                   (domain_config, persistent)
               in
               let create_info = generate_create_info ~xc ~xs vm persistent in
-              let domid = Domain.make ~xc ~xs create_info domain_config (uuid_of_vm vm) in
+              let build_info =
+                match persistent.VmExtra.build_info with
```
Comment (robhoes) [inline xc/xenops_server_xen.ml]: I think that for a VM start, the `build_info` will not be present here. It is currently constructed in `build_domain_exn`, just before calling `Domain.build`, and then stored in the `persistent` record in the local DB.

Comment (robhoes) [reply xc/xenops_server_xen.ml]: If all you need is vcpus, then you can probably just get that from `vm.vcpu_max`.

Comment (andyhhp) [reply xc/xenops_server_xen.ml]: This whole build_info vs create_info was conceptually broken in `xend`, and remains so in libxl and Xapi.  In this case, all that matters is vm.vcpu_max, which appears to be passed into `build_domain_exn`, so comes from elsewhere.

**Pull request merged**